### PR TITLE
Fixed performance drop when activating gradient clipping

### LINKED ISSUE #1522: Performance drop when activating gradient clipping
Hello all,

I experienced a substantial drop in computation time when activating gradient clipping (by passing a non-zero value to the keyword argument `gradient_clip_val` when initializing the Trainer).

I noticed that in the current implementation of the `clipping_gradient` method in pytorch-lightning/trainer/training_tricks.py redundant computations are made by first computing the 2-norm and second squaring this result, which could be shortened by computing the sum of squares directly. This saves one square root and squaring operation per parameter set.

Best,
Jonas

### Environment

```
cuda:
	GPU:
	available:           False
	version:             None
packages:
	numpy:               1.18.1
	pyTorch_debug:       False
	pyTorch_version:     1.4.0
	pytorch-lightning:   0.7.4-dev
	tensorboard:         2.2.1
	tqdm:                4.45.0
system:
	OS:                  Darwin
	architecture:
		64bit
		
	processor:           i386
	python:              3.8.2
	version:             Darwin Kernel Version 19.4.0: Wed Mar  4 22:28:40 PST 2020; root:xnu-6153.101.6~15/RELEASE_X86_64
```

### Additional context

I trained a relatively small (two-layered) MLP on MNIST; perhaps this performance drop does not become that apparent when training on larger network architectures.

### PR DESCRIPTION
**Pull request by Jonas-Jaeger**
Removed redundant computations in clip_gradients that slowed down the gradient clipping.

Fixes #1522

# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CHANGELOG.md)?

## What does this PR do?
Fixes #1522 

## PR review    
Anyone in the community is free to review the PR once the tests have passed.

### CODE REVIEW

File: pytorch_lightning/trainer/training_tricks.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def clip_gradients(self):
                 device = parameters[0].device
                 total_norm = torch.zeros([], device=device if parameters else None)
                 for p in parameters:
-                    param_norm = p.grad.data.norm(norm_type) ** norm_type
+                    param_norm = p.grad.data.pow(norm_type).sum()
```
Comment (awaelchli) [inline pytorch_lightning/trainer/training_tricks.py:43]:
```suggestion
                    param_norm = p.grad.data.abs().pow(norm_type).sum()
```
now that is what I call a norm :)

Comment (williamFalcon) [reply pytorch_lightning/trainer/training_tricks.py:43]: the other one is general to not just the 1 or 2 norm no?

Comment (awaelchli) [reply pytorch_lightning/trainer/training_tricks.py:43]: yes for all odd integers for example. Definition of p-norm: https://en.wikipedia.org/wiki/Lp_space#The_p-norm_in_finite_dimensions

Comment (williamFalcon) [reply pytorch_lightning/trainer/training_tricks.py:43]: right, sorry I thought we were modifying the implementation for 
```Trainer(track_grad_norm)```
and so i was wondering how we would generalize to the p norm because your suggestion isn't general to the p-norm

Comment (awaelchli) [reply pytorch_lightning/trainer/training_tricks.py:43]: my suggestion is not? which case does it not cover? the inf case is covered in the if clause above.

Comment (awaelchli) [reply pytorch_lightning/trainer/training_tricks.py:43]: oh, just saw that the norm_type is hardcoded as `norm_type = float(2.0)`. That is very strange.

Comment (Borda) [reply pytorch_lightning/trainer/training_tricks.py:43]: so it was merged without the `abs(...)` ? @awaelchli @williamFalcon

Comment (Jonas-Jaeger) [reply pytorch_lightning/trainer/training_tricks.py:43]: > oh, just saw that the norm_type is hardcoded as `norm_type = float(2.0)`. That is very strange.

Right, this is why I omitted the .abs() call as the 2-norm is hardcoded here. However, I would also favor your more general suggestions since this would be safer if the `norm_type` will be made variable in the future.
